tests/span_tests.cpp

```diff
@@ -54,9 +55,18 @@ struct DerivedClass : BaseClass
 };
 struct AddressOverloaded
 {
+#if (__cplusplus > 201402L)
+    [[maybe_unused]]
+#elif defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunused-member-function"
```

Comment:
After finding the [local issue](#discussion_r237531050), it turns out this warning `-Wunused-member-function` is only part of the `-Weverything` set.

The warning exists only with Clang, C++14 and if specifically enabled or with `-Weverything`.
I'll remove the pragma's, if the configuration is changed to use `-Weverything` it can be disabled for C++14.
I'll leave the `[[maybe_unused]]`. As discussed with CaseyCarter, it does state the purpose of the function. And is future-proof for improved diagnostics.